proposals/11765-KFP-SDK-common-component-support/11765-KFP-SDK-common-component-support.md

```diff
@@ -0,0 +1,88 @@
+
+    * **`kfp.components.data`:**  Components for data ingestion, transformation, and validation. Conceptual examples:
+        * `load_from_gcs(...)`: Load data from Google Cloud Storage.
+        * `split_dataset(...)`: Split a dataset into training and validation sets.
+        * `validate_schema(...)`: Validate data against a schema.
+
+    * **`kfp.components.train`:** Components for model training and hyperparameter tuning.  Conceptual examples:
+        * **`kubeflow_trainer(...)`:**  Integration with Kubeflow Training Operators
+        * **`hyperparameter_tune(...)`:**  Integration with Katib for hyperparameter optimization.
+        * `xgboost.train(...)`:  Component for XGBoost training.
+
+    * **`kfp.components.serve`:** Components for model deployment and serving. Conceptual examples:
+        * **`deploy_kserve(...)`:** Integrate with KServe for model serving on Kubernetes.
+        * `register_model(...)`:  Register a model in a model registry.
+
+    * **`kfp.components.validate`:** Components for model evaluation and monitoring. Conceptual examples:
+        * `evaluate_classification_model(...)`:  Calculate standard evaluation metrics.
+        * `data_drift_detection(...)`: Detect data drift using statistical methods.
+
+Note that the conceptual examples here are for illustrative purposes, and are not meant to be the final say.
+
+* **Implementation Details:**
+
+    * **Python-Centric Design:** Components will be primarily Python function-based, leveraging the `@kfp.dsl.component` decorator for ease of use and local testing.
+    * **Base Image Utilization:** Each Python SDK component will have an acceptible default publicly available base image (for example [pytorch/pytorch](https://hub.docker.com/r/pytorch/pytorch)).
+    * **Python Dependencies** Python dependencies not in the base image will be the default list in the `packages_to_install` parameter.
+    * **Clear Documentation:** Each SDK component will have comprehensive documentation, including:
+        * Purpose and functionality
+        * Input and output specifications
+        * Dependency requirements (packages, environment variables, cluster CRDs, Secrets etc.)
+        * Usage examples
+        * Guidance on base image selection and other parameter options.
+    * **Focus on Core Kubeflow Integrations:** Prioritize components that integrate seamlessly with core Kubeflow projects like Kubeflow Training Operators, Katib, and KServe.
+    * **Focus on User Value and Simple Design** Components that make it into the Python SDK should be there because they are widely useful in MLE work. Simplicity is a virtue. Since each new component brings complexity to the SDK, it must also bring a higher level of value. It may be possible to include more narrow use case components in library extensions.
```

Comment:
```suggestion
    * **Focus on User Value and Simple Design** Components that make it into the Python SDK should be there because they are widely useful in MLE work. Simplicity is a virtue. Since each new component brings complexity to the SDK, it must also bring a high level of value. It may be possible to include more narrow use case components in library extensions.
```